server/src/domain/asset/asset.service.ts

```diff
@@ -187,26 +190,60 @@ export class AssetService {
         await this.access.requirePermission(authUser, Permission.ARCHIVE_READ, [dto.userId]);
       }
     }
+    if (dto.withPartners) {
+      if (
+        (dto.isArchived !== undefined && dto.isArchived !== false) ||
+        (dto.isTrashed !== undefined && dto.isTrashed !== false) ||
+        (dto.isFavorite !== undefined && dto.isFavorite !== false)
```

Comment:
Am I overseeing something completely or is this similar to `dto.isArchived || dto.isTrashed || dto.isFavorite`?

Reply:
That would not trigger the case when it is undefined which should throw an error.

Reply:
Undefined doesn't throw an error currently either? e.g. `dto.isArchived !== undefined`

Reply:
Yup, that condition is wrong 😎

Reply:
Oh I see lol

Reply:
Which condition is wrong?

Reply:
If you don't send isArchived=false it will return archived photos. Not sending anything doesn't throw an error, but instead returns both archived and not archived photos.

Reply:
Ah I see, fixed the issue and add in a lot more unit tests for this block of code

Reply:
hold on, something is not right